Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -654,26 +866,24 @@ struct HfTrackIndexSkimsCreator {
 
               // fill histograms
               if (fillHistograms) {
-
-                registry.get<TH1>(HIST("hvtx2_x"))->Fill(secondaryVertex2[0]);
-                registry.get<TH1>(HIST("hvtx2_y"))->Fill(secondaryVertex2[1]);
-                registry.get<TH1>(HIST("hvtx2_z"))->Fill(secondaryVertex2[2]);
-                arrMom = array{pvec0, pvec1};
+                registry.get<TH1>(HIST("hVtx2ProngX"))->Fill(secondaryVertex2[0]);
+                registry.get<TH1>(HIST("hVtx2ProngY"))->Fill(secondaryVertex2[1]);
+                registry.get<TH1>(HIST("hVtx2ProngZ"))->Fill(secondaryVertex2[2]);
+                array<array<float, 3>, 2> arrMom = {pvec0, pvec1};
                 for (int n2 = 0; n2 < n2ProngDecays; n2++) {
                   if (isSelected2ProngCand & 1 << n2) {
-                    if ((cut2ProngInvMassCandMin[n2] < 0. && cut2ProngInvMassCandMax[n2] <= 0.) || (mass2ProngHypo1[n2] >= cut2ProngInvMassCandMin[n2] && mass2ProngHypo1[n2] < cut2ProngInvMassCandMax[n2])) {
-                      mass2ProngHypo1[n2] = RecoDecay::M(arrMom, arr2Mass1[n2]);
+                    if (whichHypo2Prong[n2] == 1 || whichHypo2Prong[n2] == 3) {
```

Comment:
Use `TESTBIT`.
Also `n2` is not a very clear name for a running index. I would suggest something like `iDecay2P`.

Reply:
Replaced everywhere (as well as `n3`)